src/components/EditPartnership/EditPartnership.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box, Button } from '@material-ui/core';
+import React, { FC } from 'react';
+import { Except } from 'type-fest';
+import { PartnershipType, UpdatePartnership } from '../../api';
+import { DialogForm, DialogFormProps } from '../Dialog/DialogForm';
+import { DialogState } from '../Dialog/useDialog';
+import {
+  CheckboxesField,
+  CheckboxOption,
+  RadioField,
+  RadioOption,
+  SubmitError,
+} from '../form';
+import { PartnershipCardFragment } from '../PartnershipCard/PartnershipCard.generated';
+import {
+  useDeletePartnershipMutation,
+  useUpdatePartnershipMutation,
+} from './EditPartnership.generated';
+
+interface EditPartnershipProps {
+  state: DialogState;
+  partnership: PartnershipCardFragment;
+}
+
+interface DialogInput {
+  partnership: Except<UpdatePartnership, 'id'>;
+}
+
+export const EditPartnership: FC<EditPartnershipProps> = ({
+  state,
+  partnership,
+}) => {
+  const [updatePartnership] = useUpdatePartnershipMutation();
+  const [deletePartnership] = useDeletePartnershipMutation();
+
+  const submit: DialogFormProps<DialogInput>['onSubmit'] = async (input) => {
+    const inputWithId = {
+      partnership: { ...input.partnership, id: partnership.id },
```

Comment:
So since we talked about this I found out (maybe via @jmc637) that we can pass ID into the `initialValues` and get it back out on submit - basically like a hidden field. This seems to be the best way to go about it as it makes types easier but still doesn't add any overhead for defining a field component for the ID.

I also do this in `EditOrganization`.

Reply:
Fixed